python/ray/data/dataset.py

```diff
@@ -3416,6 +3419,12 @@ def write_datasource(
             ray_remote_args: Kwargs passed to ``ray.remote`` in the write tasks.
             write_args: Additional write args to pass to the :class:`~ray.data.Datasource`.
         """  # noqa: E501
+        warnings.warn(
+            "`write_datasource` is deprecated in Ray 2.9. Create a `Datasink` and use "
```

Comment:
We should add a link for documentation to guide user to write a `Datasink`. Assume the documentation will be added later.